public/apps/configuration/app-router.tsx

```diff
@@ -143,140 +144,154 @@ function decodeParams(params: { [k: string]: string }): any {
-                  buildBreadcrumbs={partial(setGlobalBreadcrumbs, ResourceType.users)}
-                  {...{ ...props, ...decodeParams(match.match.params) }}
-                />
-              )}
-            />
-            <Route
-              path={ROUTE_MAP.users.href}
-              render={() => {
-                setGlobalBreadcrumbs(ResourceType.users);
-                return <UserList {...props} />;
-              }}
-            />
-            <Route
-              path={ROUTE_MAP.serviceAccounts.href}
-              render={() => {
-                setGlobalBreadcrumbs(ResourceType.serviceAccounts);
-                return <ServiceAccountList {...props} />;
-              }}
-            />
-            <Route
-              path={buildUrl(ResourceType.auditLogging) + SUB_URL_FOR_GENERAL_SETTINGS_EDIT}
-              render={() => {
-                setGlobalBreadcrumbs(ResourceType.auditLogging, 'General settings');
-                return <AuditLoggingEditSettings setting={'general'} {...props} />;
-              }}
-            />
-            <Route
-              path={buildUrl(ResourceType.auditLogging) + SUB_URL_FOR_COMPLIANCE_SETTINGS_EDIT}
-              render={() => {
-                setGlobalBreadcrumbs(ResourceType.auditLogging, 'Compliance settings');
-                return <AuditLoggingEditSettings setting={'compliance'} {...props} />;
-              }}
-            />
-            <Route
-              path={ROUTE_MAP.auditLogging.href + '/:fromType?'}
-              render={(match) => {
-                setGlobalBreadcrumbs(ResourceType.auditLogging);
-                return <AuditLogging {...{ ...props, ...match.match.params }} />;
-              }}
-            />
-            <Route
-              path={ROUTE_MAP.permissions.href}
-              render={() => {
-                setGlobalBreadcrumbs(ResourceType.permissions);
-                return <PermissionList {...props} />;
-              }}
-            />
-            <Route
-              path={ROUTE_MAP.tenants.href}
-              render={() => {
-                setGlobalBreadcrumbs(ResourceType.tenants);
-                return <TenantList tabID={'Manage'} {...props} />;
-              }}
-            />
-            <Route
-              path={ROUTE_MAP.tenantsConfigureTab.href}
-              render={() => {
-                setGlobalBreadcrumbs(ResourceType.tenants);
-                return <TenantList tabID={'Configure'} {...props} />;
-              }}
-            />
-            <Route
-              path={ROUTE_MAP.getStarted.href}
-              render={() => {
-                setGlobalBreadcrumbs();
-                return <GetStarted {...props} />;
-              }}
-            />
-            <Redirect exact from="/" to={LANDING_PAGE_URL} />
-          </Switch>
-        </EuiPageBody>
-        <CrossPageToast />
-      </EuiPage>
-    </Router>
+    <DataSourceContext.Provider value={{ dataSource, setDataSource }}>
```

Comment:
Will this automatically pass `dataSource` and `setDataSource` to the props of all children?

Do you think this should be designed with multi-select in mind?

Reply:
Yes it will. But it's actually more than that as it can also go deeper than just the next level, which can be useful for example if there's multiple children or multiple layers of children - for full context here are the docs: https://react.dev/reference/react/useContext. I don't think we need to design around multi select at this point - after this change we should have everything in a central place, making refactors pretty easy. For example - with this wrapper function at the router level, we can simply wrap the dataSource initial value to be read from the url, and wrap the setDataSource function in a function to also set that same URL param. We can do those refactors when necessary.